proposals/array_set.json

```diff
@@ -0,0 +1,215 @@
+{
+    "id": "array_set",
+    "summary": "Change the content of an array (insert, remove, update)",
+    "description": "Allows to insert into, remove from or update an array.\n\nAll labels get discarded and the array indices are always a sequence of numbers with the step size of 1 and starting at 0.",
+    "categories": [
+        "arrays"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "An array.",
+            "schema": {
+                "type": "array",
+                "items": {
+                    "description": "Any data type is allowed."
+                }
+            }
+        },
+        {
+            "name": "values",
```

Comment:
Why? So with the default parameters array_set would remove an element, instead of adding/setting. That seems counter-intuitive.

Reply:
That's maybe another indication that the name "array_set" does not align well with the functionality it provides. 

From the "array_splice" mindset I would use defaults `index=0`, `length=0`, `values=[]` , so nothing happens by default (no replacement, addition nor removal)

Reply:
My preferred way would have been to add at the end by default (i.e. array.push in JS), but a variable default length is not possible thus the default to add at the beginning (we could allow negative values to "count" backward though).

Reply:
> add at the end by default

for adding an array at the end you have (what's currently called)`array_merge` already
for adding a single item at the end you could define an additional `array_append` or `array_push` (but I'm not sure that is necessary because it can be emulated as `array_merge(data, [item])`

Reply:
Good point! Then I think it's fine as it is. It has a sensible default, which is more useful than doing nothing...

Reply:
> It has a sensible default, 

I think that largely depends on the final name of the function.

if it would be `array_insert` then `index=0, length=0` is indeed a good default
but if it would be something like `array_splice`, `array_patch` or `array_modify`, then I would not even provide a default for `index`

Reply:
Oh, wait, my mistake. By default, it changes the first array element (defaults are index = 0, length = 1). I guess it's fine to make index required. Adding before/after an array both are tackled by array_merge, too.

Reply:
Made index required.